src/resolve_type.hack

```diff
@@ -17,7 +17,57 @@ function resolve_type(
   Script $root,
   Node $node,
 ): shape('kind' => ResolvedTypeKind, 'name' => string) {
-  $uses = Resolution\get_current_uses($root, $node);
+  $autoimports = keyset[
+    // from hhvm/hphp/hack/src/parser/namespaces.ml
+    AsyncFunctionWaitHandle::class,
+    AsyncGenerator::class,
+    AsyncGeneratorWaitHandle::class,
+    AsyncIterator::class,
+    AsyncKeyedIterator::class,
+    Awaitable::class,
+    AwaitAllWaitHandle::class,
+    Collection::class,
+    ConditionWaitHandle::class,
+    Container::class,
+    ExternalThreadEventWaitHandle::class,
+    IMemoizeParam::class,
+    ImmMap::class,
+    ImmSet::class,
+    ImmVector::class,
+    InvariantException::class,
+    Iterable::class,
+    Iterator::class,
+    KeyedContainer::class,
+    KeyedIterable::class,
+    KeyedIterator::class,
+    KeyedTraversable::class,
+    Map::class,
+    ObjprofObjectStats::class,
+    ObjprofPathsStats::class,
+    ObjprofStringStats::class,
+    Pair::class,
+    RescheduleWaitHandle::class,
+    ResumableWaitHandle::class,
+    Set::class,
+    Shapes::class,
+    SleepWaitHandle::class,
+    StaticWaitHandle::class,
+    Traversable::class,
+    TypeStructure::class,
+    TypeStructureKind::class,
+    Vector::class,
+    WaitableWaitHandle::class,
+    XenonSample::class,
+    // not a legal type anymore but still supported by runtime
+    'callable',
+  ];
+
+  if (C\contains_key($autoimports, $type)) {
```

Comment:
With the exception of ‘callable’, $autoimports should be fully qualified given they’re from ::class; sorry if I’ve missed something, but it looks like:

- this will only match on qualified names
- we shouldn’t be adding HH\ as its already qualified

Probably want a C\any( Str\ends_with

Reply:
You're right, I noticed it doesn't work when I tested more. I'll change them back to strings.

Reply:
The test coverage for various edge cases in hhast is not that great, but definition-finder has great coverage for these, so I end up finding most issues when I run definition-finder's tests on my changes in hhast :/